code/edgeidentity/src/main/java/com/adobe/marketing/mobile/edge/identity/EventUtils.java

```diff
@@ -169,23 +150,9 @@ static boolean isSharedStateUpdateFor(final String stateOwnerName, final Event e
 	 */
 	static String getAdId(final Event event) {
 		final Map<String, Object> data = event.getEventData();
-		String adID;
-
-		try {
-			adID = (String) data.get(IdentityConstants.EventDataKeys.ADVERTISING_IDENTIFIER);
-		} catch (ClassCastException e) {
-			MobileCore.log(
-				LoggingMode.DEBUG,
-				LOG_TAG,
-				"EventUtils - Failed to extract ad ID from event, expected String: " + e.getLocalizedMessage()
-			);
-			return "";
-		}
+		final String adID = DataReader.optString(data, IdentityConstants.EventDataKeys.ADVERTISING_IDENTIFIER, "");
 
-		if (adID == null || IdentityConstants.Default.ZERO_ADVERTISING_ID.equals(adID)) {
-			return "";
-		}
-		return adID;
+		return (adID == null || IdentityConstants.Default.ZERO_ADVERTISING_ID.equals(adID) ? "" : adID);
```

Comment:
We should have adID set to `""` as it is the fallback value right? So we can simply return adID here

Reply:
Not sure I completely follow _simply return adID here_ !. We'd still need the comparison with  `ZERO_ADVERTISING_ID` isn't it?

Also, you made me realise that I actually intended to make the fallback `null` instead of `""` (especially when there is some additional check being done on what the optString returns. It is not be relevant here, but it may be ambiguous for the additional check to find out if the value was indeed `""` or was transformed to `""` due to it being `null`). 
Do you still prefer to fallback to `""` or, falling back to null?

Reply:
Yes we need to check ZERO string. Let's fallback to `null`.